src/flashmint/builders/leveraged-extended.test.ts

```diff
@@ -0,0 +1,247 @@
+    }
+    const builder = new LeveragedExtendedTransactionBuilder(provider)
+    const tx = await builder.build(buildRequest)
+    expect(tx).toBeNull()
+  })
+
+  test('returns null for invalid request (invalid swap data input/output token - no paths)', async () => {
+    const buildRequest = createBuildRequest()
+    buildRequest.swapDataInputOutputToken = {
+      exchange: 1,
+      path: [],
+      fees: [],
+      pool: '0x0000000000000000000000000000000000000000',
+    }
+    const builder = new LeveragedExtendedTransactionBuilder(provider)
+    const tx = await builder.build(buildRequest)
+    expect(tx).toBeNull()
+  })
+
+  test('returns null for invalid request (invalid swap data input/output token - univ3 fees)', async () => {
+    const buildRequest = createBuildRequest()
+    buildRequest.swapDataInputOutputToken = {
+      exchange: 3,
+      path: ['', '', ''],
+      // For UniV3 fees.length has to be path.length - 1
+      fees: [3000],
+      pool: '0x0000000000000000000000000000000000000000',
+    }
+    const builder = new LeveragedExtendedTransactionBuilder(provider)
+    const tx = await builder.build(buildRequest)
+    expect(tx).toBeNull()
+  })
+
+  test('returns null for invalid request (invalid path - exchange type none)', async () => {
+    const buildRequest = createBuildRequest()
+    buildRequest.swapDataInputOutputToken = {
+      exchange: 0,
+      path: [],
+      fees: [500],
+      pool: '0x00000000000000000000000000000000000000',
+    }
+    const builder = new LeveragedExtendedTransactionBuilder(provider)
+    const tx = await builder.build(buildRequest)
+    expect(tx).toBeNull()
+  })
+
+  test('returns tx for correct swap data with exchange type none', async () => {
+    const buildRequest = createBuildRequest()
+    buildRequest.swapDataInputOutputToken = {
+      exchange: 0,
+      path: [],
+      fees: [500],
+      pool: '0x0000000000000000000000000000000000000000',
+    }
+    const builder = new LeveragedExtendedTransactionBuilder(provider)
+    const tx = await builder.build(buildRequest)
+    expect(tx).not.toBeNull()
+  })
+
+  test('returns a tx for minting ETH2X (ERC20)', async () => {
+    const buildRequest = createBuildRequest()
+    buildRequest.isMinting = true
+    const indexToken = buildRequest.outputToken
+    const refTx = await contract.populateTransaction.issueExactSetFromERC20(
+      indexToken,
+      buildRequest.outputTokenAmount,
+      buildRequest.inputToken,
+      buildRequest.inputTokenAmount,
+      buildRequest.swapDataDebtCollateral,
+      buildRequest.swapDataInputOutputToken
+    )
+    const builder = new LeveragedExtendedTransactionBuilder(provider)
+    const tx = await builder.build(buildRequest)
+    if (!tx) fail()
+    expect(tx.to).toBe(FlashMintLeveragedExtendedAddress)
+    expect(tx.data).toEqual(refTx.data)
+  })
+
+  test('returns a tx for minting ETH2X (ETH)', async () => {
+    const buildRequest = createBuildRequest(true, eth, 'ETH')
```

Comment:
should this be the 2x token? (similar to L200)

Reply:
Since this is minting outputToken is ETH2X which is the default: https://github.com/IndexCoop/flash-mint-sdk/pull/48/files#diff-63922e4c38d75ebb21208c533ae7d4a5bb2b0fd0c1bba0f80bb9ecc1a3d37a45R226. I was lazy with the request function 😅